src/Init/Lean/Elab/Term.lean

```diff
@@ -649,12 +649,13 @@ when (namedArgs.any $ fun namedArg' => namedArg.name == namedArg'.name) $
   throwError ref ("argument '" ++ toString namedArg.name ++ "' was already set");
 pure $ namedArgs.push namedArg
 
-private def resolveLocalNameAux (lctx : LocalContext) : Name → List String → Option (Expr × List String)
-| n@(Name.str pre s _), projs =>
+private partial def resolveLocalNameAux (lctx : LocalContext) : Name → List String → Option (Expr × List String)
+| n, projs =>
   match lctx.findFromUserName? n with
   | some decl => some (decl.toExpr, projs)
-  | none      => resolveLocalNameAux pre (s::projs)
-| _, _ => none
+  | none      => match n with
+    | Name.str pre s _ => resolveLocalNameAux pre (s::projs)
+    | _                => none
```

Comment:
@leodemoura I think we'll have to finetune the interplay between macro-tagged identifiers and projection notation later on for the new frontend, but this change at least ensures that `scp.0` as produced by quotations can be looked up at all.

Reply:
@leodemoura FWIW, this is what currently happens if you try to elaborate projection notation coming from a quotation:
```
`(let discr := $discr; if discr.isOfKind $(quote shape.fst) && Array.size (Syntax.getArgs discr) == $(quote shape.snd) then $yes else $no)
...
error: unknown identifier 'discr.isOfKind.0'
```
Because the quotation cannot know what parts of a local reference are projections, it appends `.0` to both `discr` and `discr.isOfKind` instead of producing the "correct" `discr.0.isOfKind`. I believe we should have a general API
```
mangleIdent : Name -> List MacroScope -> Name
demangleIdent : Name -> Name \x List MacroScope
```
Then we should change both `resolveLocalName` and `resolveGlobalName` to first demangle the name, then do the recursion, and query the lctx/env with the remangled name. That is, given `a.b.c.1.2.3`, it would look up
```
a.b.c.1.2.3
a.b.1.2.3.c
a.1.2.3.b.c
```
With the mangling API, we can also transparently introduce the mangling prefix we discussed. Btw, we might also want to use `demangleIdent` on identifiers in any user-facing output like messages.
Does this make sense?

Reply:
(You *can*, however, already use them in antiquotations since these are not macro-scoped)

Reply:
I think the `mangeIdent` and `demangleIdent` approach is good.